src/ParametricOptInterface.jl

```diff
@@ -818,25 +843,100 @@ function MOI.optimize!(model::ParametricOptimizer)
         end
         empty!(model.updated_parameters)
     end
+
     MOI.optimize!(model.optimizer)
+
+    if model.evaluate_duals
+        calculate_dual_of_parameters(model)
+    end
 end
 
 ### Duals
 
-function MOI.get(model::ParametricOptimizer, attr::MOI.ConstraintDual, ci::MOI.ConstraintIndex)
-    return MOI.get(model.optimizer, attr, ci)
+function calculate_dual_of_parameters(model::ParametricOptimizer) 
+    param_dual_cum_sum = Dict{Int, Float64}()
+
+    for vi in keys(model.parameters)
+        param_dual_cum_sum[vi.value] = 0.0
+    end
+    
+    # TODO
+    # Quadratic constraints
+    for S in ScalarLinearSetList
+        affine_constraint_cache_inner = model.affine_constraint_cache[MOI.ScalarAffineFunction{Float64}, S{Float64}]
+
+        if length(affine_constraint_cache_inner) > 0
+            param_dual_cum_sum = iterate_over_constraint_cache_affine(model, affine_constraint_cache_inner ,param_dual_cum_sum)
+        end
+
+        # for F in [MOI.ScalarQuadraticFunction]
+        #     affine_constraint_cache_inner = model.affine_constraint_cache[F{Float64}, S{Float64}]
+
+        #     if length(affine_constraint_cache_inner) > 0
+        #         param_dual_cum_sum = iterate_over_constraint_cache_quadratic(model, affine_constraint_cache_inner ,param_dual_cum_sum)
+        #     end
+        # end
+    end
+
+    for param in model.affine_objective_cache
+        vi = param.variable_index.value
+        param_dual_cum_sum[vi] += param.coefficient
+    end
```

Comment:
separate into a function `update_duals_in_affine_objective` is a suggestion for a name

Reply:
Same as above, don't you think we could check if the dict `model.affine_objective_cache` is empty?